Metalama.Framework.Engine/CodeModel/DerivedTypeIndex.NamedType.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
+
+using Metalama.Framework.Code;
+using Microsoft.CodeAnalysis;
+using System.Collections.Generic;
+
+namespace Metalama.Framework.Engine.CodeModel;
+
+public sealed partial class DerivedTypeIndex
+{
+    /// <summary>
+    /// Represents either a <see cref="INamedTypeSymbol"/> or <see cref="INamedType"/>.
+    /// </summary>
+    private readonly struct NamedType
```

Comment:
Rename to TypeModelOrSymbol.

Reply:
I removed this and refactored the index to use only Ref<INamedType>.